benchmarks/benchmarks.py

```diff
@@ -235,38 +263,242 @@ def setup(self, *args):
 
 
 class TotalLocationEventsSuite:
+    params, param_names = make_params(
+        {
+            "level": ["cell", "admin3"],
+            "interval": ["day", "min"],
+            "direction": ["out", "both"],
+        }
+    )
     timer = timeit.default_timer
-    params = [list(set(x)) for x in zip(*FLOWDB_CONFIGS)] + [
-        ["cell", "admin3"],
-        ["day", "min"],
-        ["out", "both"],
-    ]
-    param_names = FLOWDB_CONFIG_PARAM_NAMES + ["level", "interval", "direction"]
     timeout = 1200
+    version = 0
 
     def setup(self, *args):
-        self.tle = TotalLocationEvents(
+        self.query = TotalLocationEvents(
             "2016-01-01",
             "2016-01-07",
             level=args[-3],
             interval=args[-2],
             direction=args[-1],
         )
-        self.tle.turn_off_caching()
+        self.query.turn_off_caching()
 
     def time_total_location_events(self, *args):
-        _ = self.tle.store().result()
+        _ = self.query.store().result()
 
     def track_total_location_events_cost(self, *args):
-        return self.tle.explain(format="json")[0]["Plan"]["Total Cost"]
+        return self.query.explain(format="json")[0]["Plan"]["Total Cost"]
+
+
+class HartiganClusterSuite:
+    params, param_names = make_params(
+        {"hours": [(4, 17), "all"], "radius": [0.1, 10.0]}
+    )
+    timer = timeit.default_timer
+    timeout = 1200
+    version = 0
+
+    def setup(self, *args):
+        self.query = subscriber_location_cluster(
+            "hartigan", "2016-01-01", "2016-01-07", hours=args[-2], radius=args[-1]
+        )
+        self.query.turn_off_caching()
+
+    def time_hartigan_cluster(self, *args):
+        _ = self.query.store().result()
+
+    def track_hartigan_cluster_cost(self, *args):
+        return self.query.explain(format="json")[0]["Plan"]["Total Cost"]
+
+
+class EventScoreSuite:
+    params, param_names = make_params(
+        {"level": ["versioned-cell", "admin3"], "hours": [(4, 17), "all"]}
+    )
+    timer = timeit.default_timer
+    timeout = 1200
+    version = 0
+
+    def setup(self, *args):
+        self.query = EventScore(
+            start="2016-01-01", stop="2016-01-07", level=args[-2], hours=args[-1]
+        )
+        self.query.turn_off_caching()
+
+    def time_event_score(self, *args):
+        _ = self.query.store().result()
+
+    def track_event_score_cost(self, *args):
+        return self.query.explain(format="json")[0]["Plan"]["Total Cost"]
+
+
+class MeaningfulLocationsSuite:
+    params, param_names = make_params(
+        {"label": ["day", "unknown"], "caching": [True, False]}
+    )
+    timer = timeit.default_timer
+    timeout = 1200
+    version = 0
+
+    def setup(self, *args):
+        hc = subscriber_location_cluster(
+            "hartigan", "2016-01-01", "2016-01-07", radius=1.0
+        )
+        es = EventScore(start="2016-01-01", stop="2016-01-07", level="versioned-site")
+        if args[-1]:
```

Comment:
How about an explicit var name here (for better readability)?
```suggestion
        do_caching = args[-1]
        if do_caching:
```